jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunSettingsEditor.kt

```diff
@@ -8,37 +8,33 @@ import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.options.SettingsEditor
 import com.intellij.openapi.project.Project
 import software.aws.toolkits.core.credentials.CredentialProviderNotFound
+import software.aws.toolkits.core.region.AwsRegion
 import software.aws.toolkits.core.utils.getLogger
 import software.aws.toolkits.core.utils.warn
 import software.aws.toolkits.jetbrains.core.AwsResourceCache
 import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
-import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
 import software.aws.toolkits.jetbrains.services.lambda.resources.LambdaResources
+import java.util.concurrent.atomic.AtomicReference
 import javax.swing.JPanel
 
 class RemoteLambdaRunSettingsEditor(project: Project) : SettingsEditor<RemoteLambdaRunConfiguration>() {
     private val view = RemoteLambdaRunSettingsEditorPanel(project)
     private val credentialManager = CredentialManager.getInstance()
-    private val regionProvider = AwsRegionProvider.getInstance()
     private val resourceCache = AwsResourceCache.getInstance(project)
+    private val settings = AtomicReference<Pair<AwsRegion?, String?>>(null to null)
+    private val functionName = AtomicReference<String?>()
 
-    init {
-        view.credentialSelector.setCredentialsProviders(credentialManager.getCredentialProviders())
-        view.regionSelector.setRegions(regionProvider.regions().values.toList())
-
-        view.credentialSelector.addActionListener { updateFunctions() }
-        view.regionSelector.addActionListener { updateFunctions() }
-    }
-
-    private fun updateFunctions() {
+    internal fun updateFunctions(region: AwsRegion?, credentialProvider: String?) {
+        val (oldRegion, oldCredentialProvider) = settings.getAndUpdate { (_, _) -> region to credentialProvider }
+        if (oldRegion == region && oldCredentialProvider == credentialProvider) return
         view.functionNames.selectedItem = null
         view.setFunctionNames(emptyList())
 
-        val credentialProviderId = view.credentialSelector.getSelectedCredentialsProvider() ?: return
-        val region = view.regionSelector.selectedRegion ?: return
+        credentialProvider ?: return
```

Comment:
If credentialProvider or region are null, would you want to clear out `functionName` before returning?

Reply:
ported to the `ResourceSelector` now we don't need any of this logic - it's all handled for us